services/inventory/agents.go

```diff
@@ -289,3 +289,76 @@ func (as *AgentsService) Remove(ctx context.Context, id string) error {
 	}
 	return nil
 }
+
+// AgentFilters represents filters for agents list.
+type AgentFilters struct {
+	// Return only Agents running on that Node.
+	RunsOnNodeID string
+	// Return only Agents that provide insights for that Node.
+	NodeID string
+	// Return only Agents that provide insights for that Service.
+	ServiceID string
+}
+
+func agentsByFilters(q *reform.Querier, filters AgentFilters) ([]*models.AgentRow, error) {
+	var conditions []string
+	var args []interface{}
+	if filters.RunsOnNodeID != "" {
```

Comment:
Since only one filter can be enabled, those ifs can be simplified to switch:
```
switch {
case filters.RunsOnNodeID != "":
  // ...
case filters.NodeID != "":
  // ...
//...
}
```